hey-discount/src/main/webapp/resources/js/View.js

```diff
@@ -0,0 +1,589 @@
+        } //i don't know why but it works only like this
+        //template.querySelector('.ad-description').textContent = ad.description;
+        block.querySelector('.discount-amount').textContent = ad.discount;
+        block.querySelector('.validUntil').textContent = ad.validUntil.getMonth() + '/' + ad.validUntil.getDate() + '/' + ad.validUntil.getFullYear();
+        if (ad.photoLink) {
+            block.querySelector('.photoLink').setAttribute('src', ad.photoLink);
+        } else {
+            block.querySelector('.photoLink').setAttribute('src', 'resources/images/no-pic.jpg');
+        }
+        block.querySelector('.vendor').textContent = ad.vendor;
+        var select = document.getElementById('vendorSelect');
+        var option = document.createElement('option');
+        option.value = ad.vendor;
+        option.text = ad.vendor;
+        select.add(option, null);
+        block.querySelector('.more').setAttribute('href', ad.link);
+        block.querySelector('.hashtags').textContent = '#' + ad.hashTags.join(' #'); 
+        /*if (ad.rating >= 1 && ad.rating < 2) {
+            block.getElementById('1star').className += ' checked'; 
+        } else if (ad.rating >= 2 && ad.rating < 3) {
+            block.getElementById('1star').className += ' checked'; 
+            block.getElementById('2star').className += ' checked'; 
+        } else if (ad.rating >= 3 && ad.rating < 4) {
+            block.getElementById('1star').className += ' checked'; 
+            block.getElementById('2star').className += ' checked'; 
+            block.getElementById('3star').className += ' checked'; 
+        } else if (ad.rating >= 4 && ad.rating < 5) {
+            block.getElementById('1star').className += ' checked'; 
+            block.getElementById('2star').className += ' checked'; 
+            block.getElementById('3star').className += ' checked'; 
+            block.getElementById('4star').className += ' checked'; 
+        } else if (ad.rating == 5) {
+            block.getElementById('1star').className += ' checked'; 
+            block.getElementById('2star').className += ' checked'; 
+            block.getElementById('3star').className += ' checked'; 
+            block.getElementById('4star').className += ' checked';
+            block.getElementById('5star').className += ' checked';
+        }*/ //it should work but i don't know why it doesn't :(
+        if (!ad.reviews) {
+            block.querySelector(".review-1").style.borderColor = 'white';
+            block.querySelector(".review-2").style.borderColor = 'white';
+            block.querySelector(".review-3").style.borderColor = 'white';
+            block.querySelector(".review-3").textContent = 'no reviews yet!';
+            block.querySelector(".review-3").style.textAlign = 'center';
+            block.querySelector(".review-3").style.fontSize = '100%';
+            block.querySelector(".review-3").style.color ='#9fb1fe';
```

Comment:
`querySelector` operation (as well as `getElementById` etc) affects page performance, so in case you need to use some part of markup, let's create a variable which will store value.